saleor/payment/gateways/np_atobarai/api.py

```diff
@@ -0,0 +1,263 @@
+import logging
+from typing import Dict, List, Optional, Tuple, Union
+
+from ....order.models import Fulfillment, Order
+from ...interface import PaymentData
+from ...models import Payment
+from .api_helpers import (
+    cancel,
+    format_price,
+    get_goods_with_discount,
+    get_refunded_goods,
+    handle_unrecoverable_state,
+    np_request,
+    register,
+    report,
+)
+from .api_types import (
+    ApiConfig,
+    PaymentResult,
+    PaymentStatus,
+    error_payment_result,
+    errors_payment_result,
+)
+from .const import PRE_FULFILLMENT_ERROR_CODE
+from .errors import (
+    FULFILLMENT_REPORT,
+    NO_PSP_REFERENCE,
+    PAYMENT_DOES_NOT_EXIST,
+    TRANSACTION_CANCELLATION,
+    TRANSACTION_CHANGE,
+    TRANSACTION_REGISTRATION,
+    add_action_to_code,
+    get_error_messages_from_codes,
+)
+from .utils import get_shipping_company_code, np_atobarai_opentracing_trace
+
+logger = logging.getLogger(__name__)
+
+
+def register_transaction(
+    order: Optional[Order], config: ApiConfig, payment_information: "PaymentData"
+) -> PaymentResult:
+    """Create a new transaction in NP Atobarai.
+
+    On pending status from NP the transaction is cancelled and
+    reason for pending is returned as error message.
+    """
+    with np_atobarai_opentracing_trace("np-atobarai.checkout.payments.register"):
```

Comment:
Maybe I am wrong but I would say that the tracing should be only on the request, not request+processing and resopnse. 
@fowczarek  Can you confirm?